lib/src/core/models/user_preferences.dart

```diff
@@ -0,0 +1,17 @@
+import "package:freezed_annotation/freezed_annotation.dart";
+import "package:isar/isar.dart";
+import "package:kana_to_kanji/src/core/constants/languages.dart";
+
+part "user_preferences.g.dart";
+
+@embedded
+@JsonSerializable()
+class UserPreferences {
+  @enumValue
+  final Languages language;
```

Comment:
Instead of using a new enum language, why not save a `Locale`?

Reply:
I haven't think about it hehe

Reply:
Isar doesn't like it : `Unsupported type. Please add @embedded to the type or implement toJson() and fromJson() methods or annotate the property with @ignore let Isar to ignore it.`

I tried to use extensions, JsonConverter, but it seems to have an issue to convert a Locale for Isar. So, what do you want to do ? Try to force something and write custom code without Freezed ?
Also maybe I missed something which can be done to resolve this issue

Reply:
mmh 🤔 je vais faire un test, je te reviens la dessus aujourd'hui